string part_type
---
bool success
string message
